website/docs/getting-started/conversation-patterns.ipynb

```diff
@@ -0,0 +1,989 @@
+     "text": [
+      "Summarize the takeaway from the conversation. Do not add any introductory phrases.\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(ConversableAgent.DEFAULT_summary_prompt)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also use a custom prompt by setting the `summary_prompt` argument\n",
+    "of `initiate_chat`.\n",
+    "\n",
+    "There are some other useful information in the `chat_result` object, including\n",
+    "the conversation history, human input, and token cost."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[{'content': 'What is triangle inequality?', 'role': 'assistant'}, {'content': 'Triangle inequality theorem is a fundamental principle in geometry that states that the sum of the lengths of any two sides of a triangle must always be greater than the length of the third side. In a triangle with sides of lengths a, b, and c, the theorem can be written as:\\n\\na + b > c\\na + c > b\\nb + c > a\\n\\nEach of these represents the condition for one specific side (a, b, or c). All must be true for a triangle to exist.', 'role': 'user'}, {'content': 'Thank you for the explanation. This theorem helps in understanding the basic properties of a triangle. It can also be useful when solving geometric problems or proving other mathematical theorems. Can you give me an example of how we can use the triangle inequality theorem?', 'role': 'assistant'}, {'content': 'Absolutely! Here\\'s an example:\\n\\nSuppose you\\'re given three line segments with lengths 10, 7, and 3 units. The question is: \"Can these three line segments form a triangle?\"\\n\\nTo answer this, you would use the triangle inequality theorem. Adding any two side lengths together should be greater than the third:\\n\\n- For sides 10 and 7: 10 + 7 = 17, which is larger than 3.\\n- For sides 10 and 3: 10 + 3 = 13, which is larger than 7.\\n- For sides 7 and 3: 7 + 3 = 10, which is equal to the length of the third side (10), but not greater.\\n\\nSo, these three lines cannot form a triangle, because not all pairs of sides satisfy the triangle inequality theorem.', 'role': 'user'}]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Get the chat history.\n",
+    "print(chat_result.chat_history)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "({'total_cost': 0.04521, 'gpt-4-0613': {'cost': 0.04521, 'prompt_tokens': 709, 'completion_tokens': 399, 'total_tokens': 1108}}, {'total_cost': 0})\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Get the cost of the chat.\n",
+    "print(chat_result.cost)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Sequential Two-Agent Chats\n",
```

Comment:
Shall we put this subsection after "Group chat", and also rename it to "Sequential Chats". "Two-Agent Chats" sounds kind of restricted. In sequential chats, each chat could involve more than two agents (although on the outer level, there are always two agents, i.e., the recipient and the sender). For example, the recipient/sender agent could be a group chat manager which will involve a set of agents. Another example is when the recipient/sender has nested chats, it involves more than two agents.

Reply:
I changed the name to "Sequential Chats". In the Group Chat section, I added a new subsection that talks about group chat as part of a sequential chat. The reason I put Sequential Chats as the second section is because it is a natural extension of the two-agent chat -- the methods are super alike, so logically it seems more coherent to have them together. The new section on group chat as part of a sequential chat should give reader some idea how different patterns can be composed. Furthermore, the nested chat section (not done yet) should give another two examples on nesting a sequential chat and nesting a group chat.